tests/system/test_dbapi.py

```diff
@@ -328,6 +328,53 @@ def test_DDL_autocommit(shared_instance, dbapi_database):
     conn.commit()
 
 
+@pytest.mark.skipif(_helpers.USE_EMULATOR, reason="Emulator does not support json.")
+def test_autocommit_with_json_data(shared_instance, dbapi_database):
+    """Check that DDLs in autocommit mode are immediately executed for
+    json fields."""
+    # Create table
+    conn = Connection(shared_instance, dbapi_database)
+    conn.autocommit = True
+
+    cur = conn.cursor()
+    cur.execute(
+        """
+        CREATE TABLE JsonDetails (
+            DataId     INT64 NOT NULL,
+            Details    JSON,
+        ) PRIMARY KEY (DataId)
+    """
+    )
+    conn.close()
```

Comment:
You probably don't need to `close()` connection and open new for every operation. It should be okay to use single `Connection` and `Cursor` objects for the whole test.

Reply:
removed extra connection creation and reused the 1st connection.